### DISCUSSION TITLE
v1.98.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#7713] [Announcements]
# v1.98.0

## Announcement

The [logo design contest](https://github.com/immich-app/immich/discussions/6318) has ended on March 4th, 2024. 

We had an internal voting round with the core team members and contributors. The winner is @tyree, with the design below.

![Tyree's design](https://github.com/immich-app/immich/assets/27055614/c6d2b295-bd55-4797-8add-2b93bdb7da35)

We are planning to roll out the new logo in the near future.

Thank you all for your contributions and discussions. We are very happy to have such strong community engagement.

## Highlights

Welcome to release `v1.98.0` of Immich. We are pleased to introduce many new features, quality of life improvements and bug fixes in this release—some of the highlights below.

- Move library settings to the Administration page - Simplify the external library setup process.
- Context search with people filtering.
- File name and file extension search.
- Add an option to disable the required password change when creating a new user.
- Show people in the information sheet on the mobile app.
- A configurable timer is used to delete a user permanently.

### Move library settings to the Administration page

We moved all library settings under the Admin settings to reduce the complexity of setting up external libraries and lay the groundwork for sharing external libraries between users. 

![image](https://github.com/immich-app/immich/assets/27055614/1116a52d-700d-4c43-9521-4c5b6efdcc3e)

### Context search with people filtering

Thanks to @michelheusschen's help in solving the technical challenges of combining contextual search results with people filtering, it is now possible to search for "person in red dress" and choose only the people that you would like to see in a red dress. 

### File name and extension search

You can now search for file name and file extension again as free text search. For example, you can search for all Nikon photos with `NEF` or `nef`, or search for file names such as `IMG_1234.JPG` with `123` or `234` as the search query.

![image](https://github.com/immich-app/immich/assets/27055614/40816acb-f4a3-41cd-b7da-7ff1c045a7af)

### Show people the information sheet on the mobile app.

You can now see the people detected in a photo/video in the detail sheet on the mobile app.

<img width="400" src="https://github.com/immich-app/immich/assets/27055614/a6581bb3-c280-488d-b9fd-e8df78fc094c"/>

### Option to disable required password change for new user creation

![image](https://github.com/immich-app/immich/assets/27055614/00f0a8fa-c376-4ea0-bb19-43d39ada7b90)

### Configurable timer for user deletion

This setting can be found in the `server settings` in the `Administration page`. You can now adjust how long until a user gets permanently removed from the server. 

![image](https://github.com/immich-app/immich/assets/27055614/0764ade3-87d7-40ff-b81e-47bb34e25ee2)

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 3QVAb9dCHutquVejeNXitPqZX26Yg5kxb7

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### 🗄️ Server
* fix(server): re-add mimalloc by @mertalev in https://github.com/immich-app/immich/pull/7511
* fix(server): only queue ml / transcoding jobs after thumbnail generation on upload by @mertalev in https://github.com/immich-app/immich/pull/7516
* feat: filter people when using smart search by @michelheusschen in https://github.com/immich-app/immich/pull/7521
* fix(immich-admin): only 1st argument was passed by @saimonn in https://github.com/immich-app/immich/pull/7552
* feat(server/web): add oauth defaultStorageQuota and storageQuotaClaim by @samholton in https://github.com/immich-app/immich/pull/7548
* feat(server): allow oauth claim to set 0 for no quota by @samholton in https://github.com/immich-app/immich/pull/7581
* fix(server): prevent leaking isFavorite and isArchived info by @martabal in https://github.com/immich-app/immich/pull/7580
* feat(server, web): add checkbox to create user screen for shouldChang… by @samholton in https://github.com/immich-app/immich/pull/7598
* fix(server): regular version check by @rovo89 in https://github.com/immich-app/immich/pull/7620
* fix(server, web): prevent reload when liking an asset by @martabal in https://github.com/immich-app/immich/pull/7589
* fix(server): incorrect number of assets for a person by @martabal in https://github.com/immich-app/immich/pull/7602
* feat(server): deterministic download order by @jrasm91 in https://github.com/immich-app/immich/pull/7658
* chore(server): eslint await-thenable by @etnoy in https://github.com/immich-app/immich/pull/7545
* fix(server): stack info in asset response for mobile by @alextran1502 in https://github.com/immich-app/immich/pull/7346
* feat(server,web): make user deletion delay configurable by @samholton in https://github.com/immich-app/immich/pull/7663
* chore(server): save original file name with extension by @alextran1502 in https://github.com/immich-app/immich/pull/7679
* chore(server): lower default max recognition distance for facial recognition by @mertalev in https://github.com/immich-app/immich/pull/7689
* perf(server): use queries to refresh library assets by @mertalev in https://github.com/immich-app/immich/pull/7685
* chore(server): search filename using originalFileName by @alextran1502 in https://github.com/immich-app/immich/pull/7691
* fix(server): access face count when the value is undefined by @alextran1502 in https://github.com/immich-app/immich/pull/7694
* fix(server): add extension to filename migration by @michelheusschen in https://github.com/immich-app/immich/pull/7697
* chore(server): Move library watcher to microservices by @etnoy in https://github.com/immich-app/immich/pull/7533
* fix(server): admin access to edit library by @alextran1502 in https://github.com/immich-app/immich/pull/7708
### 📱 Mobile
* fix(mobile): bottom bar Upload translation by @waclaw66 in https://github.com/immich-app/immich/pull/7553
* refactor(mobile): Refactor video player page and gallery bottom app bar by @martyfuhry in https://github.com/immich-app/immich/pull/7625
* feat(mobile): Add people list to exit bottom sheet by @l0nax in https://github.com/immich-app/immich/pull/6717
* fix(mobile): Refactors exif bottom sheet to use widgets and fixes slow sliding up exif bottom sheet by @martyfuhry in https://github.com/immich-app/immich/pull/7671
* chore(mobile,web): use originalFilename by @alextran1502 in https://github.com/immich-app/immich/pull/7692
### 🖥️ Web
* fix(web): asset disappears from album after metadata edit by @michelheusschen in https://github.com/immich-app/immich/pull/7520
* chore(web): fix eslint setup in VS Code by @benmccann in https://github.com/immich-app/immich/pull/7543
* fix(web): prevent duplicate calls to time bucket endpoint by @mertalev in https://github.com/immich-app/immich/pull/7563
* perf(web): asset delete by @michelheusschen in https://github.com/immich-app/immich/pull/7555
* fix(web): show history search box only when needed by @martabal in https://github.com/immich-app/immich/pull/7544
* feat(web): ascending order for slideshow by @martabal in https://github.com/immich-app/immich/pull/7502
* feat(web): show user quota on server stats page by @samholton in https://github.com/immich-app/immich/pull/7591
* feat(web): add share link on asset-viewer by @samholton in https://github.com/immich-app/immich/pull/7595
* feat(web): improve alt text by @mertalev in https://github.com/immich-app/immich/pull/7596
* perf(web): optimize date groups by @mertalev in https://github.com/immich-app/immich/pull/7593
* feat(web): use lib/utils/copyToClipboard for share link by @samholton in https://github.com/immich-app/immich/pull/7603
* fix(web): prettify album download filename by @waclaw66 in https://github.com/immich-app/immich/pull/7637
* Revert "perf(web): optimize date groups" by @michelheusschen in https://github.com/immich-app/immich/pull/7638
* Revert "fix(web): prevent duplicate calls to time bucket endpoint" by @alextran1502 in https://github.com/immich-app/immich/pull/7644
* feat(web): search by filename by @alextran1502 in https://github.com/immich-app/immich/pull/7624
* fix(web): date input on chrome by @michelheusschen in https://github.com/immich-app/immich/pull/7669
* perf(web): optimize response sizes for initial page load by @mertalev in https://github.com/immich-app/immich/pull/7594
* fix(web): consistent modal escape behavior by @michelheusschen in https://github.com/immich-app/immich/pull/7677
* fix(web): save filename search in search filter box by @alextran1502 in https://github.com/immich-app/immich/pull/7704
### 🧠 Machine Learning
* feat(machine-learning): support cuda 12 by @martabal in https://github.com/immich-app/immich/pull/7569
* feat: preloading of machine learning models by @DawidPietrykowski in https://github.com/immich-app/immich/pull/7540
### ⚡ CLI
* feat(cli): Use well-known endpoint to resolve API by @bo0tzz in https://github.com/immich-app/immich/pull/6733
### 📓 Documentation
* docs: fix the database name env variable by @ikajdan in https://github.com/immich-app/immich/pull/7576
* doc: update truenas.md with permission info. by @NicholasFlamy in https://github.com/immich-app/immich/pull/7606
### 🔨 Maintenance
* chore(deps): update dependency @types/pg to v8.11.1 by @renovate in https://github.com/immich-app/immich/pull/7522
* chore(deps): bump tj-actions/verify-changed-files from 18 to 19 by @dependabot in https://github.com/immich-app/immich/pull/7524
* chore: build tweaks by @jrasm91 in https://github.com/immich-app/immich/pull/7484
* refactor(web): drop axios by @jrasm91 in https://github.com/immich-app/immich/pull/7490
* chore: linting by @jrasm91 in https://github.com/immich-app/immich/pull/7532
* refactor(e2e): use better dummy assets by @jrasm91 in https://github.com/immich-app/immich/pull/7536
* feat(server,web): remove external path nonsense and make libraries admin-only by @etnoy in https://github.com/immich-app/immich/pull/7237
* refactor: library e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7538
* chore: remove unused upload property by @jrasm91 in https://github.com/immich-app/immich/pull/7535
* fix(web): immich version by @martabal in https://github.com/immich-app/immich/pull/7541
* chore(deps): update dependency @types/pg to v8.11.2 by @renovate in https://github.com/immich-app/immich/pull/7585
* chore(deps): update dependency @playwright/test to v1.42.0 by @renovate in https://github.com/immich-app/immich/pull/7601
* fix: bump version pipeline by @martabal in https://github.com/immich-app/immich/pull/7586
* chore(deps): update dependency @types/node to v20.11.22 by @renovate in https://github.com/immich-app/immich/pull/7610
* chore(deps): update dependency @types/node to v20.11.22 by @renovate in https://github.com/immich-app/immich/pull/7630
* chore(deps): update @immich/cli by @renovate in https://github.com/immich-app/immich/pull/7629
* chore(deps): update dependency prettier-plugin-svelte to v3.2.2 by @renovate in https://github.com/immich-app/immich/pull/7633
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/7635
* chore(deps): update @immich/cli by @renovate in https://github.com/immich-app/immich/pull/7640
* chore(deps): update dependency @types/node to v20.11.23 by @renovate in https://github.com/immich-app/immich/pull/7641
* chore(deps): update dependency @types/node to v20.11.24 by @renovate in https://github.com/immich-app/immich/pull/7646
* chore(cli): remove unused packages by @jrasm91 in https://github.com/immich-app/immich/pull/7648
* chore(deps): update dependency @types/node to v20.11.24 by @renovate in https://github.com/immich-app/immich/pull/7645
* chore(deps): update dependency @types/node to v20.11.24 by @renovate in https://github.com/immich-app/immich/pull/7649
* chore(deps): update dependency svelte-check to v3.6.6 by @renovate in https://github.com/immich-app/immich/pull/7650
* chore: download e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7651
* chore(deps): update server by @renovate in https://github.com/immich-app/immich/pull/7652
* fix(deps): update dependency archiver to v7 by @renovate in https://github.com/immich-app/immich/pull/7622
* chore(deps): update dependency @types/cookie-parser to v1.4.7 by @renovate in https://github.com/immich-app/immich/pull/7661
* chore(deps): bump sanitize-html, fixing CVE-2024-21501 by @etnoy in https://github.com/immich-app/immich/pull/7662
* fix(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/7634
* chore(deps): update base-image to v20240305 (major) by @renovate in https://github.com/immich-app/immich/pull/7682
* refactor: e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7703

## New Contributors
* @samholton made their first contribution in https://github.com/immich-app/immich/pull/7548
* @ikajdan made their first contribution in https://github.com/immich-app/immich/pull/7576
* @DawidPietrykowski made their first contribution in https://github.com/immich-app/immich/pull/7540
* @NicholasFlamy made their first contribution in https://github.com/immich-app/immich/pull/7606

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.97.0...v1.98.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.98.0'>v1.98.0</a>.</em>

**Comment by Chaphasilor** [comment]
Amazing. I've had "gripes" with many of the things that were improved with this update in the past, so this update really adds a lot of value for me!  
Thanks for all of your work, and congrats on the new logo. But I'm curious, which font will be used in the end?

**Comment by desmond206x** [comment]
The new logo is looking great!
Thanks for the update. Showing the people faces in the mobile is very nice.

**Comment by daniele-athome** [comment]
Packed with features! And a very nice brand new logo.
Thank you!! I hope I'll be able to contribute again to this great project soon.

**Comment by rosscullen** [comment]
Thank goodness for new logo!

**Comment by Shisharo** [comment]
Great update as always. Just a bit sad about the logo change since it became a bit generic and lost the personal touch and meaning it had with Alex's love for Japanese culture

**Comment by aviv926** [comment]
Thanks for the update.
Congratulations to the winners!

With the new update, an external library is "imposed" on the user without being able to cancel it, am I right?

**Reply by etnoy** [reply]
Not sure what you mean by "imposing", we moved the library settings to only be managed by the admin. This means we can simplify the system by reducing the complexity of the external path settings.

**Reply by aviv926** [reply]
Before this update the user had the option to choose whether to import the external library or not according to his wish, now he does not have the option to choose whether to import the library or not, it just happens if the admin chooses it to happen.

Plus, the user can't even know it happened.

Maybe we should add a selection option like:
The user {Admin.user} wants to share an external library with you {path+libary name}
Do you want to add the external directory to your timeline?
Yes No

**Reply by jrasm91** [reply]
I think that is the idea moving forward. Specifically being able to share the same library with multiple users. We can take into consideration an "invite" process maybe, although shared albums work the same way (no invite).

**Reply by aviv926** [reply]
> I think that is the idea moving forward. Specifically being able to share the same library with multiple users. We can take into consideration an "invite" process maybe, although shared albums work the same way (no invite).

Well even an option to choose whether to appear in the timeline or not for a specific library would be good, because albums by default don't appear in the timeline (and they can't either).

**Reply by jrasm91** [reply]
Yeah, it makes sense. The admin had to help the user setup the external library to begin with so we just decided to have them do all of the setup instead. In the future we hope to add more options and flexibility around how they are used, which I think will address the concerns you are raising here, which are valid.

**Reply by etnoy** [reply]
As you can tell, we've been improving libraries almost every release. File watching, removing the external path headaches, performance etc. In my own opinion, there is so much potential we can still unlock with libraries. Browsing specific libraries and excluding them from the timeline is on the roadmap, but as always, no ETA. I really want to get to per-library sharing, too, but this might require some additional work on the underlying architecture.

**Reply by Chaphasilor** [reply]
Sharing libraries with multiple users would be really useful indeed. I just imported 150k family photos, but it's not super useful since they're tied to one specific account.

Family sharing is currently a bit hard to do, but I'm sure you'll get there, you're making great progress already!

**Comment by seanmuth** [comment]
New logo is great!

**Comment by othyn** [comment]
Love the new logo, it very much modernises it! It will fit much nicer alongside my other home screen apps now :D

**Comment by fscheps** [comment]
These updates look awesome!
One comment, the demo site looks completely empty: https://demo.immich.app/photos

**Comment by bcsteeve** [comment]
I had no problem with the old logo, but I gotta say the new one is pretty sharp!  Well done.

**Comment by eil047** [comment]
Great! Thanks a lot for the search on filename! 😊
